enterprise/models.py

```diff
@@ -1906,6 +1906,40 @@ def __repr__(self):
         return self.__str__()
 
 
+class SubsidizedEnterpriseCourseEnrollment(TimeStampedModel):
```

Comment:
You might want to extract most of `LicensedEnterpriseCourseEnrollment` into a base class that this thing can inherit from, too.  Like `enterprise_course_enrollment` and `is_revoked` are common to each, and we'll likely want the methods that already exist on `LicensedEnterpriseCourseEnrollment`.
Another thought is, do we want some concrete model like `EnterpriseSubsidyFulfillment` in here that `LicensedEnterpriseCourseEnrollment`, `SubsidizedEnterpriseCourseEnrollment`, and `SubsidizedEnterpriseCourseEntitlement` all inherit from?  It'd be a little squishy since two of those would have a `transaction_id` field and the license one would have `license_uuid`, but we could make that work.
There's probably other ways to model this in a more consistent way, too.
Me, you and @johnnagro could talk through this in a more structured way if you want.  This is kind of our chance to "get it right from the start".